.github/workflows/linting.yml

```
@@ -0,0 +1,22 @@
+name: Linting
+
+on:
+  pull_request:
+  push:
+    branches: [ main ]
+    tags: [ '*' ]
+  workflow_dispatch:
+
+jobs:
+  linting:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/setup-python@v5
```

Do you even need to use `setup-python`? If so, pick the latest v3 to suppress the warning:
```suggestion
      - uses: actions/setup-python@v5
        with:
          python-version: '3.x'
```